examples/advanced/xipfs/main.c

```diff
@@ -0,0 +1,231 @@
+ * > exec
+ * 2025-01-14 09:48:49,572 # exec
+ * 2025-01-14 09:48:49,573 # Hello World!
+ * > ls /dev/nvme0p0
+ * 2025-01-14 09:48:59,997 # ls /dev/nvme0p0
+ * 2025-01-14 09:48:59,999 # hello-world.fae       896 B
+ * 2025-01-14 09:49:00,000 # total 1 files
+ * > vfs df
+ * 2025-01-14 09:49:04,957 # vfs df
+ * 2025-01-14 09:49:04,962 # Mountpoint              Total         Used    Available     Use%
+ * 2025-01-14 09:49:04,968 # /dev/nvme0p0           40 KiB        4 KiB       36 KiB      10%
+ * 2025-01-14 09:49:04,974 # /dev/nvme0p1           60 KiB          0 B       60 KiB       0%
+ * execute /dev/nvme0p0/hello-world.fae ipsum dolores it
+ * 2025-01-14 09:49:14,223 # execute /dev/nvme0p0/hello-world.fae Lorem ipsum dolor sit amet
+ * 2025-01-14 09:49:14,225 # Hello World!
+ * 2025-01-14 09:49:14,225 # Lorem
+ * 2025-01-14 09:49:14,226 # ipsum
+ * 2025-01-14 09:49:14,226 # dolor
+ * 2025-01-14 09:49:14,227 # sit
+ * 2025-01-14 09:49:14,227 # amet
+ * ```
+ */
+int execution_handler(int argc, char **argv) {
+    (void)argc;
+    (void)argv;
+
+    int file_handle = vfs_open(FILENAME_OF_HELLO_WORLD_FAE, O_RDONLY, 0);
+    if (file_handle < 0) {
+
+        /** There's no executable file yet, let's drop one */
+        int ret = xipfs_extended_driver_new_file(
+            FILENAME_OF_HELLO_WORLD_FAE, SIZEOF_HELLO_WORLD_FAE, 1
+        );
+        if (ret < 0) {
+            printf("xipfs_extended_driver_new_file : failed to create '%s' : error=%d\n",
+                   FILENAME_OF_HELLO_WORLD_FAE, ret);
+            return EXIT_FAILURE;
+        }
+
+        /**
+         * Fill it with blob data
+         * Take care : vfs does not support O_APPEND with vfs_write, only O_WRONLY or O_RDWR
+         */
+        file_handle = vfs_open(FILENAME_OF_HELLO_WORLD_FAE, O_WRONLY, 0);
+        if (file_handle < 0) {
+            printf("vfs_open : failed to open '%s' : error =%d\n",
+                   FILENAME_OF_HELLO_WORLD_FAE, file_handle);
+            return EXIT_FAILURE;
+        }
+
+        ssize_t write_ret = vfs_write(file_handle, hello_world_fae, SIZEOF_HELLO_WORLD_FAE);
+        if (write_ret < 0) {
+            printf("vfs_write : failed to fill '%s' : error=%d\n",
+                   FILENAME_OF_HELLO_WORLD_FAE, write_ret);
+            vfs_close(file_handle);
+            return EXIT_FAILURE;
+        }
+    }
+
+    vfs_close(file_handle);
+
+    char *exec_argv[] = {
+        FILENAME_OF_HELLO_WORLD_FAE,
+        NULL
+    };
+
+
+    int ret = xipfs_extended_driver_execv(FILENAME_OF_HELLO_WORLD_FAE, exec_argv);
+    if (ret < 0) {
+        printf("Failed to execute '%s' : error=%d\n", FILENAME_OF_HELLO_WORLD_FAE, ret);
+        return EXIT_FAILURE;
+    }
+
+    return EXIT_SUCCESS;
+}
+
+static shell_command_t shell_commands[] = {
+    {"exec", "Execute Hello World", execution_handler},
+    {NULL, NULL, NULL},
+};
+
+#else // BOARD_DWM1001
+
+#define shell_commands NULL
```

Comment:
```suggestion
static shell_command_t shell_commands[] = { {NULL, NULL, NULL} };
```

I think this might be a cleaner solution. I have to try if your current solution would work or not, but it's better to actually define the right data type for the variable.